### PR TITLE
tidb_query: fix parse 'yyyy-mm-dd HH:MM:SS' format to Duration

### PR DESCRIPTION
**Pull request by iosmanthus**
###  What have you changed?

This PR fixes the issue that cannot parse `yyyy-mm-dd HH:MM:SS` to duration. The TiDB and MySQL can handle this format properly.

```
mysql> select convert('2006-05-13 03:14:14', time);
+--------------------------------------+
| convert('2006-05-13 03:14:14', time) |
+--------------------------------------+
| 03:14:14                             |
+--------------------------------------+
1 row in set (0.10 sec)
```

Don't assume reviewers understand the original issue.

###  What is the type of the changes?

- Bugfix (a change which fixes an issue)

###  How is the PR tested?

- Unit test

### CODE REVIEW

File: components/tidb_query/src/codec/mysql/duration.rs [hunk 1]
```diff
@@ -361,21 +417,7 @@ impl Duration {
...
-                    let mut ctx = EvalContext::default();
-                    let dt =
-                        DateTime::parse_datetime(&mut ctx, integer_part, fsp as i8, true).ok()?;
-                    dt.convert(&mut ctx).ok()
-                } else {
-                    None
-                }
-            })
-            .ok_or_else(|| Error::truncated_wrong_val("TIME", input))
+            .ok_or_else(|| Error::truncated_wrong_val("TIME", format!("{:?}", input)))
```
[the remark this answers is not in the document]
Comment (iosmanthus) [reply components/tidb_query/src/codec/mysql/duration.rs]: Sorry, forgot to revert it.

File: components/tidb_query/src/codec/mysql/duration.rs [hunk 2]
```diff
@@ -140,6 +140,56 @@ mod parser {
         Ok((rest, hhmmss))
     }
 
+    fn hhmmss_datetime(input: &str, fsp: u8) -> IResult<&str, [u32; 4], ()> {
+        let (rest, digits) = digit1(input)?;
+        if digits.len() == 12 || digits.len() == 14 {
```
Comment (lonng) [inline components/tidb_query/src/codec/mysql/duration.rs]: I think we shouldn't assume the length of digital string and it's the duty of `DateTime::parse_datetime`.

**Comment by andylokandy** [comment]
@sre-bot /merge

**Pull request merged**